pontoon/pretranslation/tasks.py

```diff
@@ -173,6 +173,9 @@ def pretranslate(self, project_pk, locales=None, entities=None):
     changed_entities = {}
     existing = ChangedEntityLocale.objects.values_list("entity", "locale").distinct()
     for t in translations:
+        if t.entity.resource.project.data_source == Project.DataSource.DATABASE:
```

Comment:
We should add project to the [`prefetch_related()`](https://docs.djangoproject.com/en/4.1/ref/models/querysets/#prefetch-related) call [above](https://github.com/mozilla/pontoon/blob/f3727dd6afe8d621d61d2a4311818d5781dc517d/pontoon/pretranslation/tasks.py#L68), which will retrieve project data from the DB in one call, instead of doing that for each translation separately. DB queries are slow, so we should reduce them.

Note that you can simply replace `resource` with `resource__project` in the call above.